### PR TITLE
Updated tropical tree cover layer | Toggle switch implementation

### PR DESCRIPTION
**Pull request by SARodrigues**
## IMPORTANT

https://github.com/Vizzuality/gfw/pull/4517/commits/bf4bfb1139d777b4da34c2cf8aa4e7a627a4cf0a needs to be removed before this PR is merged. It is only included at the moment to facilitate testing for both FE and Science. 

## Overview

This PR adds the possibility of using a toggle/switch in the map legend, in order to switch dataset layers. 

**In this PR:**
- Added the possibility to use a toggle switch to switch between a dataset's layers
  - It is enabled when the `isToggleLayer` property is set on a dataset  
  - Added a new `light-alternate` theme to the existing `<Switch />` component  
  - The `<Switch />` component can now also be displayed in a smaller size  
- Updated `gfw-components` to the latest version  
- Fixed an error _"state mutation detected between dispatches' error when switching layers"_
  _only is shown in development, made it impossible to switch layers_  
- Fixed an issue with `statementConfig` (layer statement) where the statement was built based on a dataset's default layer  
  _the correct configuration is to show the statement on a per layer basis_  
- Fixed an issue with the `<LayerStatement />` component, in which the `className` was not being applied for simple statements  
- Minor auto-linting fixes

## Screenshot

Screenshot 2023-03-03 at 15 28 13

## Test Environment

https://gfw-staging-pr-4517.herokuapp.com/

## Tracking

[FLAG-698](https://gfw.atlassian.net/browse/FLAG-698)

### CODE REVIEW

File: services/datasets.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const environmentString = () => {
 export const getDatasets = () =>
   rwRequest
     .get(
-      `/dataset?application=gfw&includes=metadata,vocabulary,layer&published=true&page[size]=9999&env=${environmentString()}${
+      `/dataset?application=gfw&includes=metadata,vocabulary,layer&published=true&page[size]=9999&env=production,preproduction,staging${
```
Comment (SARodrigues) [inline services/datasets.js:18]: This needs to be reverted before the PR is merged.

File: providers/datasets-provider/actions.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ const handleFeatureEnvLock = (env) => {
     return true;
   }
   if (currEnv === 'preproduction') {
-    return ['preproduction', 'production'].includes(env);
+    return ['preproduction', 'production', 'staging'].includes(env);
```
Comment (SARodrigues) [inline providers/datasets-provider/actions.js:23]: This needs to be reverted before the PR is merged.

File: components/map/components/legend/index.js [hunk 3]
```diff
@@ -90,13 +90,10 @@ class Legend extends PureComponent {
   onChangeLayer = (layerGroup, newLayerKey) => {
     const { setMapSettings, activeDatasets } = this.props;
     setMapSettings({
-      datasets: activeDatasets.map((l) => {
-        const dataset = l;
-        if (l.dataset === layerGroup.dataset) {
-          dataset.layers = [newLayerKey];
```
Comment (SARodrigues) [inline components/map/components/legend/index.js:96]: This was the reason for the `state mutation detected between dispatches` error.

**Review by pedrogpimenta — CHANGES_REQUESTED** [review]
- [x] Also, a reminder to remove the forced env variables from the request :)

File: components/map/components/legend/component.jsx [hunk 4]
```diff
@@ -198,20 +212,23 @@ const MapLegend = ({
...
                   )}
                 {(isSelectorLayer || isMultiSelectorLayer) &&
+                  !isToggleLayer &&
                   selectorLayerConfig && (
                     <LayerSelectorMenu
                       className="layer-selector"
                       layerGroup={lg}
                       name={name}
                       multi={isMultiSelectorLayer}
+                      toggle={isToggleLayer}
```
[the remark this answers is not in the document]
Comment (SARodrigues) [reply components/map/components/legend/component.jsx:231]: Good catch! You're right, it is not needed here; it'll be false inside the component. 
I'll remove it :)

**Pull request merged**